collect_app/src/main/java/org/odk/collect/android/preferences/source/SettingsProvider.kt

```diff
@@ -20,7 +18,7 @@ class SettingsProvider(private val context: Context) {
     }
 
     @JvmOverloads
-    fun getGeneralSettings(projectId: String = NOT_SPECIFIED_UUID): Settings {
+    fun getGeneralSettings(projectId: String? = null): Settings {
         val settingsId = getSettingsId(GENERAL_SETTINGS_NAME, projectId)
 
         return settings.getOrPut(settingsId) {
```

Comment:
Can't we improve this by removing that `SharedPreferencesSettings(PreferenceManager.getDefaultSharedPreferences(context), GeneralKeys.DEFAULTS)`
If you migrated settings now it should be needed only to read those default settings during migration and once this operation is done we shouldn't use it right? It should work like in the case of adminSettings. If I'm right you could remove it from here to avoid confusion and use that `getDefaultSharedPreferences` directly in the code where you perform that migration or have a separate method here in this class that would return default settings but has a comment or a good descriptive name.

Reply:
Ah, yeah nice idea! The project importer currently grabs the default shared prefs and admin prefs directly (as it knows that's exactly what it needs) so we can remove that complexity here now.